spec/facility_factory_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+require 'rspec'
+
+describe FacilityFactory do
+  describe '#create_facilities' do
+    it 'can create facilities from external data (Colorado)' do
+      factory = FacilityFactory.new
+      co_dmv_office_locations = DmvDataService.new.co_dmv_office_locations
+      colorado_facilities = factory.create_facilities(co_dmv_office_locations)
+
+      facility_zero = colorado_facilities[0]
+      expect(facility_zero).to be_a(Facility)
+      expect(facility_zero.name).to eq("DMV Tremont Branch")
```

Comment:
These are very robust tests! The only risk here is that the facility data is coming from an external service (via the DmvDataService) so there's a slight chance that this data could change/update and cause your tests to fail even if there's nothing wrong with your code or the way you're handling their data. This is a risk that is *totally* fine for the project, but as you progress in Turing (especially Mod 3) you'll start to think about other options when you're testing something that uses a data set we can't control. Two other options are:
1. using more general assertions (i.e. expect(facility_zero.name).to be_a String)
2. using mock (dummy) data instead of actually getting data from the DmvDataService

This is food for thought for as you progress through the program. For now, these are great tests!

Reply:
Thanks! I had been wondering how to deal with external database changes, but this clears some of that up for me for sure.